### ISSUE TITLE
websocket support

### ISSUE THREAD

**Issue by drselump14** [mana-ethereum/ethereumex#62]
Is there any plan to include support for websocket?
I saw there is PR for this, but unfortunately the owner has removed it.
If necessary I have time to contribute to it. Please give me a guidance

**Comment by hswick** [comment]
I believe the reasoning is that in practice implementing this isn't that helpful.

One reason is that having a timer for retrieving event filter changes every 1 second with a http request, works perfectly fine for production Ethereum apps. They deal with high network latency anyways.

You can take a look at how I do this with my library:
https://github.com/hswick/exw3#events

Combine this idea with basic Elixir processes (GenServer, etc) and you have all the concurrency you wanted from the websocket implementation.

This is a simpler implementation and less error prone. It also is fairly trivial to integrate it with other tools like Phoenix channels.

**Comment by InoMurko** [comment]
@hswick what do you mean that in practice `implementing this isn't that helpful`?

**Comment by InoMurko** [comment]
@drselump14 if you need an example how a subscription via web sockets would look like: https://github.com/omisego/potterhat/pull/20/files#diff-2b9cdc401d068615be9a97ae309d295e

**Comment by hswick** [comment]
@InoMurko `isn't that helpful` may be a bit too harsh. I'm sure one can imagine cases where they would like to use it.

However, I still stand by my point that push v pull is only an incremental improvement. As I mentioned a 1 second lag is generally acceptable for Ethereum applications. In the case you mentioned, this would decrease the load on the client by 50%. Unless the Ethereum network has gotten exponentially faster in the past 6 months, then this point hasn't changed.

TLDR: the gains provided by the websocket may not outweigh the complexities it brings about.
